app/src/main/java/com/android/ubclaunchpad/driver/UI/sessionInfoActivity.java

```diff
@@ -27,89 +33,126 @@ protected void onCreate(Bundle savedInstanceState) {
         final ListView listView = (ListView) findViewById(R.id.sessionItemsList);
         final ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, android.R.id.text1, itemsArray);
         listView.setAdapter(adapter);
-        String sessionName = getIntent().getStringExtra(getString(R.string.session_name));
+        final String sessionName = getIntent().getStringExtra(getString(R.string.session_name));
         final String passengerDistance = "\nP\n\t\t\t\t";
         final String driverDistance = "\nD\n\t\t\t\t";
         TextView textViewSessionName = (TextView) findViewById(R.id.viewSessionName);
         textViewSessionName.setText(sessionName);
 
-        //Adding Drivers
-        FirebaseUtils.getDatabase().child("Session Group").child(sessionName).child("drivers").addChildEventListener(new ChildEventListener() {
+        final DatabaseReference session = FirebaseUtils.getDatabase()
+                .child("Session group")
+                .child(sessionName);
+        final String UID = FirebaseUtils.getFirebaseUser().getUid();
+        //find current user's id
+        FirebaseUtils.getDatabase()
+                .child(StringUtils.FirebaseUserEndpoint)
+                .child(UID)
+                .addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot userSnapshot) {
+                        final User currentUser = userSnapshot.getValue(User.class);
```

Comment:
It looks like you're getting the User currently using the app, any reason for getting it from Firebase rather than UserManager? Getting it from Firebase gives us more nested callbacks, which we'd want to avoid if possible.